content/5_Plat/RURQ.mdx

```diff
@@ -221,9 +221,171 @@ queries.
 	</Resource>
 </Resources>
 
-### Implementation
+### Solution - Range Updates & Sums
 
-<IncompleteSection />
+<Spoiler title="Solution">
+
+This question asks you to support the following types of queries. 
+
+- Add a value to all elements within the range $[a,b]$
+
+- Set all values within the range $[a,b]$ to a certain value 
+
+- Find the sum of all values in range $[a,b]$
+
+Consider the first two types of queries, a lazy tag will be created in each node of the tree for each type. In this solution, $lz1$ will represent the lazy tag for the range add query and $lz2$ will represent the lazy tag for the range set query. 
+
+Given the two different update queries, a total of four different situations might take place as a result of any update. 
+
+- Range add ($lz1$) when $lz2$ equals 0: Simply add the new value to the pre-existing value 
+
+- Range add ($lz1$) when $lz2$ doesnt equal to 0: Add the new value to $lz2$ and clear $lz1$ 
```

Comment:
> doesnt equal to 0

"is not equal to 0" or "doesn't equal 0" or $lz2\neq 0$ preferred

Reply:
Addressed.

Reply:
Not addressed.

Reply:
Should be good now.

Reply:
Not addressed.